lib/measured/rails/validations.rb

```diff
@@ -6,13 +14,40 @@ def validate_each(record, attribute, measurable)
     measurable_unit = record.send("#{ attribute }_unit")
     measurable_value = record.send("#{ attribute }_value")
 
-    message = options[:message] || "is not a valid unit"
+    return unless measurable_unit.present? || measurable_value.present?
+
+    record.errors.add(attribute, message) if [measurable_unit.blank?, measurable_value.blank?].any?
+
+    record.errors.add(attribute, message) unless measured_class.valid_unit?(measurable_unit)
 
     if options[:units]
       valid_units = [options[:units]].flatten.map{|u| measured_class.conversion.to_unit_name(u) }
       record.errors.add(attribute, message) unless valid_units.include?(measured_class.conversion.to_unit_name(measurable_unit))
     end
 
-    record.errors.add(attribute, message) unless measured_class.valid_unit?(measurable_unit)
+    options.slice(*CHECKS.keys).each do |option, value|
+      record.errors.add(attribute, message) unless measurable.send(CHECKS[option], value_for(value, record))
```

Comment:
This one is less important is it comes from an internal hash of operations, but you're right and it can't hurt.